src/internals/ir/callGraph.ts

```diff
@@ -0,0 +1,196 @@
+import { unreachable } from "../util";
+import { TactASTStore } from "./astStore";
+import { IdxGenerator } from "./indices";
+import { MistiContext } from "../../";
+import { Logger } from "../../internals/logger";
+import { forEachExpression } from "../tact/iterators";
+import {
+  AstFunctionDef,
+  AstReceiver,
+  AstContractInit,
+  AstExpression,
+  AstMethodCall,
+  AstStaticCall,
+} from "@tact-lang/compiler/dist/grammar/ast";
+
+type CGNodeId = number & { readonly brand: unique symbol };
+type CGEdgeId = number & { readonly brand: unique symbol };
+
+/**
+ * Represents an edge in the call graph, indicating a call from one function to another.
+ */
+class CGEdge {
+  public idx: CGEdgeId;
+
+  constructor(
+    public src: CGNodeId,
+    public dst: CGNodeId,
+  ) {
+    this.idx = IdxGenerator.next("cg_edge") as CGEdgeId;
+  }
+}
+
+/**
+ * Represents a node in the call graph, corresponding to a function or method.
+ */
+class CGNode {
+  public idx: CGNodeId;
+  public inEdges: Set<CGEdgeId> = new Set();
+  public outEdges: Set<CGEdgeId> = new Set();
+
+  constructor(
+    public astId: number | undefined,
```

Comment:
The meaning of the `undefined` value should be documented. Also, I would show a `logger.debug` message when we cannot find the relevant ASTId, since that's indicates a bug in Misti.